libs/cli/src/main/java/org/elasticsearch/cli/Terminal.java

```diff
@@ -216,5 +245,58 @@ public String readText(String text) {
         public char[] readSecret(String text) {
             return readText(text).toCharArray();
         }
+
+        @Override
+        public char[] readSecret(String text, int maxLength) {
+            getErrorWriter().println(text);
+            return readLineToCharArray(getReader(), maxLength);
+        }
+
+        /**
+         * This method has to keep track of a little bit of state in order to account for
+         * line breaks in a system-independent way.
+         *
+         * Visible for testing.
+         */
+        char[] readLineToCharArray(Reader reader, int maxLength) {
+            char[] buf = new char[maxLength];
```

Comment:
This  should be maxLength + 2 right? So we can account for a \r\n on windows being in the buffer?